### PR TITLE
New Python Solutions

### PR DESCRIPTION
**Pull request by AAkashyap1**

### CODE REVIEW

File: solutions/usaco-784.mdx [hunk 1]
```diff
@@ -64,4 +64,33 @@ int main(){
...
+```py
+file_in = open('lifeguards.in')
+data = file_in.read().strip().split('\n')
+n = int(data[0])
+
+lifeguards = [list(map(int, data[i + 1].split(' '))) for i in range(n)]
+
+max_time_covered = 0
+
+# Simulate firing each cow and computing the total time covered by the remaining life guards
```
Comment (SansPapyrus683) [inline solutions/usaco-784.mdx:78]:
```suggestion
"""
Simulate firing each cow and computing the total
time covered by the remaining life guards
"""
```

File: solutions/usaco-784.mdx [hunk 2]
```diff
@@ -64,4 +64,33 @@ int main(){
...
+# Compute the maximum of these times
+for i in range(n):
+	time_covered = 0
+	for v in range(1, 1001):
```
Comment (SansPapyrus683) [inline solutions/usaco-784.mdx:82]: no magic numbers please, maybe a `MAX_M` or something (idk i haven't read the problem) and then replacing `for v in range(1, MAX_M + 1)` would be beeter

File: solutions/usaco-832.mdx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+[Official Analysis](http://www.usaco.org/current/data/sol_milkorder_bronze_open18.html)
+
+## Solution 2
+
+In this problem, there are three cases that we have to consider. The first of these cases is that cow one is fixed. In this case, the solution is simple, as we just output its fixed position. The second case we must consider is if cow one is in the given hierarchy of cows. In this case, we want to place every cow in the hierarchy as early as possible. The last case is when cow one is neither fixed nor in the hierarchy. In this case, we would want to place every cow in the hierarchy as late as possible, allowing for cow one to get the earliest position possible.
+<LanguageSection>
+<PySection>
+
+```py
```
Comment (maggie-j-liu) [inline solutions/usaco-832.mdx:18]: use tabs instead of spaces to indent

File: solutions/usaco-832.mdx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+file_in = open('milkorder.in')
+data = file_in.read().strip().split('\n')
+n, m, k = map(int, data[0].split(' '))
+hierarchy = list(map(int, data[1].split(' ')))
+fixed = [list(map(int, data[i+2].split(' '))) for i in range(k)]
```
Comment (maggie-j-liu) [inline solutions/usaco-832.mdx:23]:
```suggestion
fixed = [list(map(int, data[i + 2].split(' '))) for i in range(k)]
```

File: solutions/usaco-784.mdx (hunk 2 above)
Comment (maggie-j-liu) [reply solutions/usaco-784.mdx:82]: idk; i think that a constant would help when the number is used multiple times, but it's only used once here so it's not as necessary?

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

**Review by dongliu0426 — CHANGES_REQUESTED** [review]
add time complexity (both sols)

File: solutions/usaco-784.mdx (hunk 2 above)
Comment (darren-yao) [reply solutions/usaco-784.mdx:82]: It's bronze, so the use of MAX_N could make the code less confusing to read, but either way I don't think it's super important.

**Review by darren-yao — CHANGES_REQUESTED** [review]
@AAkashyap1 I don't have anything further to say. Please address Dong's and Maggie's comments, and then let us know when you're done. Thanks!

File: solutions/usaco-784.mdx (hunk 2 above)
Comment (AAkashyap1) [reply solutions/usaco-784.mdx:85]: I'll add it, just in case.

File: solutions/usaco-832.mdx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+author: Ananth Kashyap
+---
+
+## Solution 1
+
+[Official Analysis](http://www.usaco.org/current/data/sol_milkorder_bronze_open18.html)
+
+## Solution 2
+
+In this problem, there are three cases that we have to consider. The first of these cases is that cow one is fixed. In this case, the solution is simple, as we just output its fixed position. The second case we must consider is if cow one is in the given hierarchy of cows. In this case, we want to place every cow in the hierarchy as early as possible. The last case is when cow one is neither fixed nor in the hierarchy. In this case, we would want to place every cow in the hierarchy as late as possible, allowing for cow one to get the earliest position possible.
```
Comment (frodakcin) [inline solutions/usaco-832.mdx:14]: It'd be nice to break this down into bullet points.

**Review by frodakcin — CHANGES_REQUESTED** [review]

File: solutions/usaco-784.mdx [hunk 6]
```diff
@@ -2,7 +2,7 @@
 id: usaco-784
 source: USACO Bronze 2018 January
 title: Lifeguards
-author: Danh Ta Chi Thanh
+author: Danh Ta Chi Thanh, Ananth Kashyap
 ---
 
 [Official Analysis (Java)](http://www.usaco.org/current/data/sol_lifeguards_bronze_jan18.html)
```
Comment (HarryWangATX) [inline solutions/usaco-784.mdx:8]:
```suggestion
[Official Analysis](http://www.usaco.org/current/data/sol_lifeguards_bronze_jan18.html)
```

File: solutions/usaco-784.mdx [hunk 7]
```diff
@@ -64,4 +64,37 @@ int main(){
 
 </CPPSection>
 
+<PySection>
+
```
Comment (HarryWangATX) [inline solutions/usaco-784.mdx:68]:
```suggestion

## Python Implementation

```

File: solutions/usaco-832.mdx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+<LanguageSection>
+<PySection>
+
```
Comment (HarryWangATX) [inline solutions/usaco-832.mdx:17]:
```suggestion

## Python Implementation

```

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

**Comment by AAkashyap1** [comment]
> @AAkashyap1 I don't have anything further to say. Please address Dong's and Maggie's comments, and then let us know when you're done. Thanks!

@darren-yao I believe I have addressed everything! Thank you guys for the comments and suggestions.

File: solutions/usaco-832.mdx [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+---
+
+## Solution 1
+
+[Official Analysis](http://www.usaco.org/current/data/sol_milkorder_bronze_open18.html)
+
+## Solution 2
+
+In this problem, there are three cases that we have to consider: 
+- The first of these cases is that cow one is fixed. In this case, the solution is simple, as we just output its fixed position.
```
Comment (dongliu0426) [inline solutions/usaco-832.mdx:15]:
```suggestion
- The first of these cases is when cow one is fixed. In this case, the solution is simple, as we just output its fixed position. 
```

File: solutions/usaco-832.mdx [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+## Solution 1
+
+[Official Analysis](http://www.usaco.org/current/data/sol_milkorder_bronze_open18.html)
+
+## Solution 2
+
+In this problem, there are three cases that we have to consider: 
+- The first of these cases is that cow one is fixed. In this case, the solution is simple, as we just output its fixed position. 
+- The second case we must consider is if cow one is in the given hierarchy of cows. In this case, we want to place every cow in the hierarchy as early as possible.
```
Comment (dongliu0426) [inline solutions/usaco-832.mdx:16]:
```suggestion
- In the second case, we must consider is if cow one is in the given hierarchy of cows. In this case, we want to place every cow in the hierarchy as early as possible. 
```

File: solutions/usaco-832.mdx [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+## Python Implementation
+
+```py
+file_in = open('milkorder.in')
+data = file_in.read().strip().split('\n')
+n, m, k = map(int, data[0].split(' '))
+hierarchy = list(map(int, data[1].split(' ')))
+fixed = [list(map(int, data[i + 2].split(' '))) for i in range(k)]
+
+cows = [0 for i in range(n)]
```
Comment (dongliu0426) [inline solutions/usaco-832.mdx:30]: `cows = [0] * n`

File: solutions/usaco-832.mdx [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+
+for i in range(k):
+  	cows[fixed[i][1] - 1] = fixed[i][0]
+
+if 1 in cows:
```
Comment (dongliu0426) [inline solutions/usaco-832.mdx:35]: can you add comments here? like `# case 1/2/3`

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

**Comment by dongliu0426** [comment]
> add time complexity (both sols)

also this has not been addressed @AAkashyap1

File: solutions/usaco-832.mdx (hunk 12 above)
Comment (AAkashyap1) [reply solutions/usaco-832.mdx:35]: Will do

**Comment by AAkashyap1** [comment]
> > add time complexity (both sols)
> 
> also this has not been addressed @AAkashyap1

Sorry my bad

File: solutions/usaco-784.mdx [hunk 13]
```diff
@@ -64,4 +67,39 @@ int main(){
 
 </CPPSection>
 
+<PySection>
```
Comment (dongliu0426) [inline solutions/usaco-784.mdx:70]: different idea, so can you put this in a separate `<LanguageSection>` with `# Solution 2` or smth?

File: solutions/usaco-832.mdx [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+- The first of these cases is when cow one is fixed. In this case, the solution is simple, as we just output its fixed position. 
+- In the second case, we must consider is if cow one is in the given hierarchy of cows. In this case, we want to place every cow in the hierarchy as early as possible. 
+- The last case is when cow one is neither fixed nor in the hierarchy. In this case, we would want to place every cow in the hierarchy as late as possible, allowing for cow one to get the earliest position possible.
+
+**Time Complexity:** $\mathcal{O}(N(M + N))$
+
+<LanguageSection>
+<PySection>
+
+## Python Implementation
```
Comment (maggie-j-liu) [inline solutions/usaco-832.mdx:24]: move this outside languagesection (and above the time complexity) and just put ## Implementation

@HarryWangATX i think you suggested this? the template on the guide just has ## Implementation, so i think we should stick with that
i think the reason is that the header should be above time complexity

File: solutions/usaco-784.mdx [hunk 15]
```diff
@@ -64,4 +67,39 @@ int main(){
...
+
+## Python Implementation
```
Comment (maggie-j-liu) [inline solutions/usaco-784.mdx:72]: same thing, move outside languagesection (and delete ## C++ Implementation from the cppsection)

File: content/2_Bronze/Ad_Hoc.problems.json [hunk 16]
```diff
@@ -25,6 +25,18 @@
         "kind": "internal"
       }
     },
+    {
```
Comment (maggie-j-liu) [inline content/2_Bronze/Ad_Hoc.problems.json:28]: this problem was not originally in this module, so let's not add it here... can you remove this and put it in extraProblems.json instead

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: solutions/usaco-784.mdx [hunk 17]
```diff
@@ -64,4 +67,39 @@ int main(){
...
+
+```py
+file_in = open('lifeguards.in')
+data = file_in.read().strip().split('\n')
+n = int(data[0])
+
+lifeguards = [list(map(int, data[i + 1].split(' '))) for i in range(n)]
+
+MAX_N = 1000
```
Comment (maggie-j-liu) [inline solutions/usaco-784.mdx:81]: this variable is for time, right? `MAX_T` might be better in that case? or even `MAX_TIME`

File: solutions/usaco-832.mdx (hunk 14 above)
Comment (AAkashyap1) [reply solutions/usaco-832.mdx:24]: What should I do

Comment (dongliu0426) [reply solutions/usaco-832.mdx:24]: > move this outside languagesection (and above the time complexity) and just put ## Implementation

File: content/2_Bronze/Ad_Hoc.problems.json (hunk 16 above)
Comment (SansPapyrus683) [reply content/2_Bronze/Ad_Hoc.problems.json:28]: what's the harm in doing it

File: solutions/usaco-784.mdx [hunk 18]
```diff
@@ -64,4 +67,39 @@ int main(){
...
+## Python Implementation
+
+```py
+file_in = open('lifeguards.in')
+data = file_in.read().strip().split('\n')
+n = int(data[0])
+
+lifeguards = [list(map(int, data[i + 1].split(' '))) for i in range(n)]
+
+MAX_TIME = 1000
```
Comment (SansPapyrus683) [inline solutions/usaco-784.mdx:81]: put this line at the top of the file

File: solutions/usaco-784.mdx [hunk 19]
```diff
@@ -64,4 +67,39 @@ int main(){
...
+max_time_covered = 0
+
+"""
+Simulate firing each cow and computing the total
+time covered by the remaining life guards
+"""
+# Compute the maximum of these times
```
Comment (SansPapyrus683) [inline solutions/usaco-784.mdx:88]:
```suggestion
```

File: solutions/usaco-784.mdx [hunk 20]
```diff
@@ -64,4 +67,39 @@ int main(){
...
+
+```py
+file_in = open('lifeguards.in')
+data = file_in.read().strip().split('\n')
+n = int(data[0])
+
+lifeguards = [list(map(int, data[i + 1].split(' '))) for i in range(n)]
+
+MAX_TIME = 1000
+max_time_covered = 0
```
Comment (SansPapyrus683) [inline solutions/usaco-784.mdx:82]: put this between the multiline comment & the for loop

File: solutions/usaco-832.mdx [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+author: Ananth Kashyap
+---
+
+## Solution 1
+
+[Official Analysis](http://www.usaco.org/current/data/sol_milkorder_bronze_open18.html)
+
+## Solution 2
+
+In this problem, there are three cases that we have to consider:
```
Comment (SansPapyrus683) [inline solutions/usaco-832.mdx:14]: maybe do it like this
```
there are 3 cases blah blah blah
1. Cow one is fixed...
2. Cow one is in the govern hierarchy of cows...
3. Cow one is neither fixed nor in the hiearchy
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-832.mdx [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+---
+
+## Solution 1
+
+[Official Analysis](http://www.usaco.org/current/data/sol_milkorder_bronze_open18.html)
+
+## Solution 2
+
+In this problem, there are three cases that we have to consider: 
+- The first of these cases is when cow one is fixed. In this case, the solution is simple, as we just output its fixed position.
```
Comment (jessechoe10) [inline solutions/usaco-832.mdx:15]:
```suggestion
- When the first cow is fixed, we output its position. 
```

File: solutions/usaco-832.mdx [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+
+## Solution 1
+
+[Official Analysis](http://www.usaco.org/current/data/sol_milkorder_bronze_open18.html)
+
+## Solution 2
+
+In this problem, there are three cases that we have to consider: 
+- The first of these cases is when cow one is fixed. In this case, the solution is simple, as we just output its fixed position. 
+- In the second case, we must consider is if cow one is in the given hierarchy of cows. In this case, we want to place every cow in the hierarchy as early as possible.
```
Comment (jessechoe10) [inline solutions/usaco-832.mdx:16]:
```suggestion
- When cow 1 is in the given hierarchy of the cows, we should place every cow in the hierarchy as early as possible.
```

Comment (jessechoe10) [reply solutions/usaco-832.mdx:16]: I think you should elaborate on this.

File: solutions/usaco-832.mdx [hunk 24]
```diff
@@ -0,0 +1,92 @@
...
+## Solution 1
+
+[Official Analysis](http://www.usaco.org/current/data/sol_milkorder_bronze_open18.html)
+
+## Solution 2
+
+In this problem, there are three cases that we have to consider: 
+- The first of these cases is when cow one is fixed. In this case, the solution is simple, as we just output its fixed position. 
+- In the second case, we must consider is if cow one is in the given hierarchy of cows. In this case, we want to place every cow in the hierarchy as early as possible. 
+- The last case is when cow one is neither fixed nor in the hierarchy. In this case, we would want to place every cow in the hierarchy as late as possible, allowing for cow one to get the earliest position possible.
```
Comment (jessechoe10) [inline solutions/usaco-832.mdx:17]:
```suggestion
- When cow 1 is neither fixed nor in the hierarchy, we want to place every cow in the hierarchy as late as possible such that cow 1 gets the earliest possible position.
```

File: solutions/usaco-832.mdx (hunk 21 above)
Comment (frodakcin) [reply solutions/usaco-832.mdx:14]: ^ I agree

File: solutions/usaco-832.mdx (hunk 22 above)
Comment (SansPapyrus683) [reply solutions/usaco-832.mdx:15]: "When cow $1$" would be better imo

Comment (frodakcin) [reply solutions/usaco-832.mdx:15]: ^
Or even "Cow $1$ is fixed: blah blah blah"

Comment (frodakcin) [reply solutions/usaco-832.mdx:15]: Also possibly have ordered list?
```
1. blah blah blah
2. ...
3. ...
```

Unordered is ok tho.

File: solutions/usaco-832.mdx (hunk 14 above)
Comment (frodakcin) [reply solutions/usaco-832.mdx:24]:
```
## Implementation

**Time Complexity:** $\mathcal{O}(N(M + N))$

<LanguageSection>

<PySection>

... your code, etc ...
```

File: solutions/usaco-784.mdx (hunk 18 above)
Comment (frodakcin) [reply solutions/usaco-784.mdx:81]: I'd say it's personal preference, so probably not necessary

File: solutions/usaco-784.mdx (hunk 20 above)
Comment (frodakcin) [reply solutions/usaco-784.mdx:82]: Also probably personal preference.

File: solutions/usaco-784.mdx [hunk 25]
```diff
@@ -64,4 +67,39 @@ int main(){
...
+
+"""
+Simulate firing each cow and computing the total
```
Comment (frodakcin) [inline solutions/usaco-784.mdx:85]: firing each lifeguard?

File: solutions/usaco-784.mdx (hunk 13 above)
Comment (frodakcin) [reply solutions/usaco-784.mdx:70]: wait isn't this same idea ? (try firing each lifeguard?)

**Review by frodakcin — CHANGES_REQUESTED** [review]

File: solutions/usaco-784.mdx (hunk 18 above)
Comment (SansPapyrus683) [reply solutions/usaco-784.mdx:81]: by convention constants are at the top i think

**Comment by HarryWangATX** [comment]
@AAkashyap1 u here still? can you address the changes requested?

File: solutions/usaco-784.mdx (hunk 13 above)
Comment (AAkashyap1) [reply solutions/usaco-784.mdx:70]: Wait so what should I do for this

**Comment by AAkashyap1** [comment]
> @AAkashyap1 u here still? can you address the changes requested?

Is this ready to merge now?

**Review by dongliu0426 — CHANGES_REQUESTED** [review]
Change all `# C++/Java/Py Implementation` to `# Implementation` and move it out of `<LanguageSection>`

File: solutions/usaco-784.mdx [hunk 26]
```diff
@@ -2,17 +2,19 @@
...
 source: USACO Bronze 2018 January
 title: Lifeguards
-author: Danh Ta Chi Thanh
+author: Danh Ta Chi Thanh, Ananth Kashyap
 ---
 
-[Official Analysis (Java)](http://www.usaco.org/current/data/sol_lifeguards_bronze_jan18.html)
+[Official Analysis](http://www.usaco.org/current/data/sol_lifeguards_bronze_jan18.html)
+
+**Time Complexity:** $\mathcal{O}(N^2)$
```
Comment (dongliu0426) [inline solutions/usaco-784.mdx:10]: not true for your solution

Comment (frodakcin) [inline solutions/usaco-784.mdx:10]: Official solution is N * max_time runtime, not N^2

Comment (frodakcin) [reply solutions/usaco-784.mdx:10]: Something like `NT, where T is the time range`

File: solutions/usaco-784.mdx [hunk 27]
```diff
@@ -59,9 +61,45 @@ int main(){
...
 }
-
 ```
 
 </CPPSection>
+</LanguageSection>
+
+## Python Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2*\text{TIME})$
```
Comment (frodakcin) [inline solutions/usaco-784.mdx:71]: `\text{TIME}` in the time complexity is quite visually unpleasant. Prefer just saying "$\mathcal{O}(N^2*T)$, where $T$ is the maximum time" or something (just be consistent with above)

**Review by frodakcin — CHANGES_REQUESTED** [review]
Looks fine otherwise.

**Comment by dongliu0426** [comment]
> Change all `# C++/Java/Py Implementation` to `# Implementation` and move it out of `<LanguageSection>`

**Review by frodakcin — APPROVED** [review]

**Review by dongliu0426 — APPROVED** [review]
yay

**Review by darren-yao — APPROVED** [review]

**Pull request merged**